prow/config/secrets_agent.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Implements an agent to read and reload the secrets.
+package config
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// SecretAgent watches a path and automatically loads the secrets stored.
+type SecretAgent struct {
+	sync.Mutex
+	secretsMap map[string][]byte
+}
+
+// NewSecretAgent returns a SecretAgent with a map of strings with []byte value.
+// The key will be used to index the file path, and the value to store the secret value.
+func NewSecretAgent() *SecretAgent {
```

Comment:
nit: This could be combined with `Start`. I don't think there is any case in which we would construct a `SecretAgent` that we don't intend to immediately use.